### PR TITLE
[MB-3912] Drop unused confirmation_number column, replace with move code in payloads

### PR DESCRIPTION
**Pull request by duncan-truss**
## Description

The confirmation_number column wasn't being used on the orders table but was still being returned in payloads in the ghc and prime apis for MoveOrders.  This PR drops the column from the orders table and removes the field from the Orders model.

In the ghc api we include moveCode with the MoveOrder payload because this is often queried separately from the move.

In the prime api I'm adding MoveCode to the MoveTaskOrder payload because that is primarily what the prime queries as opposed to orders.  Since MoveCode is already a field on MoveTaskOrder/Move there is no need to add it to MoveOrder because you'll always have the parent MoveTaskOrder.

## Reviewer Notes

In theory an order could have many moves but this is not true in practice to my knowledge so it's okay to fetch the locator from the single move associated with the order.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make build_tools
make db_dev_e2e_populate
make server_run
make office_client_run
```

**Testing the GHC MoveOrder payload**
1. Login as the TOO user in the office app
2. Make an api request for an existing order in the database such as `http://officelocal:3000/ghc/v1/move-orders/1d49bb07-d9dd-4308-934d-baad94f2de9b`
3. The json payload response should include the moveCode property.

**Testing the Prime MoveTaskOrder payload**
1. Make an api request to the prime api to fetch data that includes a Move Task Order payload such as from your terminal `prime-api move-task-orders GET | jq .`
2. Verify that the moveCode property is part of the Move Task Order object response

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3912) for this change
* [ReDoc Prime API](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/transcom/mymove/8607a1e3eb8ab3218f6a2b5334df941a9c036ab9/swagger/prime.yaml)

## Screenshots

Screen Shot 2020-12-29 at 3 43 48 PM

Screen Shot 2020-12-30 at 1 36 58 PM

### CODE REVIEW

**Review by shimonacarvalho — CHANGES_REQUESTED** [review]
Hey Duncan, I think the support.yaml is using `locator` for this field. To make it all consistent once and for all, could you possibly change that to match the prime.yaml. 

Also if `moveCode` is to be generated by milmove not requested by the caller, could you set it to be `readOnly` in the supportapi.

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload.go [hunk 1]
```diff
@@ -109,6 +109,11 @@ func MoveOrder(moveOrder *models.Order) *ghcmessages.MoveOrder {
 		branch = ghcmessages.Branch(*moveOrder.ServiceMember.Affiliation)
 	}
 
+	var moveCode string
+	if moveOrder.Moves != nil && len(moveOrder.Moves) > 0 {
+		moveCode = moveOrder.Moves[0].Locator
+	}
```
Comment (shimonacarvalho) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:115]: Hmm this makes me think, can there be more than 1 move? If so, is there a better way to select the right one? For e.g. not a cancelled move.

Comment (duncan-truss) [reply pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:115]: Yea I mentioned that in the reviewer notes that it is in practice a 1:1 relation to my knowledge. The moves are sorted by created_at descending on the orders model so if it were possible to get more than one then we'd be getting the most recent 🤷‍♂️.

**Review by Ryan-Koch — APPROVED** [review]

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload.go (hunk 1 above)
Comment (shimonacarvalho) [reply pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:115]: Oh! Okay great.

**Review by donaldthai — APPROVED** [review]
Works and LGTM

File: swagger/prime.yaml [hunk 2]
```diff
@@ -997,6 +993,10 @@ definitions:
         example: 1f2270c7-7166-40ae-981e-b200ebdf3054
         format: uuid
         type: string
+      locator:
```
Comment (shimonacarvalho) [inline swagger/prime.yaml:996]: Sorry if I was not clear! I meant we should change `locator` to `moveCode` in both `prime.yaml` and `support.yaml` to match the rest of the system.

**Review by shimonacarvalho — CHANGES_REQUESTED** [review]

File: swagger/prime.yaml (hunk 2 above)
Comment (duncan-truss) [reply swagger/prime.yaml:996]: Haha whoops I mistook it for the reverse

Comment (shimonacarvalho) [reply swagger/prime.yaml]: Yeah I realized! So sorry. Approving now!

**Review by shimonacarvalho — APPROVED** [review]
Thanks! This will help us keep Prime in sync with rest of system.

**Pull request merged**